File: rust_exercises/leetcode/Problem739/Solution.java
class Solution {
    public int[] dailyTemperatures(int[] T) {
        int[] result = new int[T.length];
        Stack<Integer> stack = new Stack<>();
        
        for (int i = 0; i < T.length; i += 1) {
            while (!stack.isEmpty() && T[i] > T[stack.peek()]) {
                int index = stack.pop();
                result[index] = i - index;
            }
            stack.push(i);
        }
        
        return result;
    }
}

File: rust_exercises/leetcode/Problem739/solution.rs
impl Solution {
    pub fn daily_temperatures(t: Vec<i32>) -> Vec<i32> {
        let mut stack = vec![];
        let mut result = vec![0; t.len()];
        
        for (i, temp) in t.iter().enumerate() {
            while !stack.is_empty() {
                let (index, value) = stack[stack.len() - 1];
                if temp <= value {
                    break;
                }
                result[index] = (i - index) as i32;
                stack.pop();
            }
            stack.push((i, temp));
        }
        
        result
    }
}
